src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2805,6 +2822,29 @@ internal override StrongNameKeys StrongNameKeys
             return true;
         }
 
+        private void GenerateModuleInitializer(PEModuleBuilder moduleBeingBuilt, DiagnosticBag methodBodyDiagnosticBag)
+        {
+            if (GetModuleInitializerMethods() is { IsEmpty: false } moduleInitializerMethods)
+            {
+                var ilBuilder = new ILBuilder(moduleBeingBuilt, new LocalSlotManager(slotAllocator: null), OptimizationLevel.Release, areLocalsZeroed: true);
+
+                // PROTOTYPE(module-initializers): require deterministic order
+                foreach (var method in moduleInitializerMethods)
+                {
+                    ilBuilder.EmitOpCode(ILOpCode.Call, stackAdjustment: 0);
+
+                    ilBuilder.EmitToken(
+                        moduleBeingBuilt.Translate(method, methodBodyDiagnosticBag, needDeclaration: true),
+                        method.SyntaxNode,
```

Comment:
>method.SyntaxNode [](start = 24, length = 17)

```CSharpSyntaxTree.Dummy.GetRoot()```?

Reply:
Sure. If you have time, can you explain why `method.SyntaxNode` wouldn't be better? I'm not sure what this does.

Reply:
> Sure. If you have time, can you explain why method.SyntaxNode wouldn't be better? I'm not sure what this does.

I assume ```SyntaxNode``` is not directly available on MethodSymbol, we can use it right now because we are taking dependency on specific implementation of the type. I would like us to not have that dependency. Also, I do not expect us to be able to observe the difference in what syntax node we will pass. API needs it, but I doubt any new error is going to be reported at this point.

---
In reply to: [414532355](https://github.com/dotnet/roslyn/pull/43281#discussion_r414532355) [](ancestors = 414532355)